### ISSUE TITLE
Akeneo 1.7.22 : Security vulnerabilities in dependencies

### ISSUE THREAD

**Issue by kanduvisla** [akeneo/pim-community-dev#8329]
This issue has[ already been reported once for Akeneo 1.5 in the past](https://github.com/akeneo/pim-community-dev/issues/5233), but since [Akeneo 1.5 no longer has LTS](https://www.akeneo.com/knowledge-base/how-many-pim-versions-are-maintained-by-akeneo/) it might have become a lower priority.

However, Akeneo 1.7 does still have LTS up until March 2019 and these security vulnerabilities still exist. If I run the Symfony Security check I get the following:

```
Security Check Report
~~~~~~~~~~~~~~~~~~~~~

Checked file: /var/www/composer.lock


  [CRITICAL]
  8 packages have known vulnerabilities


doctrine/annotations (v1.2.6)
-----------------------------

 * CVE-2015-5723: Security Misconfiguration Vulnerability in various Doctrine projects
   https://www.doctrine-project.org/2015/08/31/security_misconfiguration_vulnerability_in_various_doctrine_projects.html

doctrine/common (v2.4.2)
------------------------

 * CVE-2015-5723: Security Misconfiguration Vulnerability in various Doctrine projects
   https://www.doctrine-project.org/2015/08/31/security_misconfiguration_vulnerability_in_various_doctrine_projects.html

doctrine/doctrine-bundle (v1.2.0)
---------------------------------

 * CVE-2015-5723: Security Misconfiguration Vulnerability in various Doctrine projects
   https://www.doctrine-project.org/2015/08/31/security_misconfiguration_vulnerability_in_various_doctrine_projects.html

doctrine/orm (v2.4.7)
---------------------

 * CVE-2015-5723: Security Misconfiguration Vulnerability in various Doctrine projects
   https://www.doctrine-project.org/2015/08/31/security_misconfiguration_vulnerability_in_various_doctrine_projects.html

dompdf/dompdf (v0.6.1)
----------------------

 * CVE-2014-5013: Remote Code Execution (complement of CVE-2014-2383)
   https://github.com/dompdf/dompdf/releases/tag/v0.6.2

 * CVE-2014-5011: Information Disclosure
   https://github.com/dompdf/dompdf/releases/tag/v0.6.2

 * CVE-2014-5012: Denial Of Service Vector
   https://github.com/dompdf/dompdf/releases/tag/v0.6.2

paragonie/random_compat (v1.4.3)
--------------------------------

 * Uses insecure CSPRNG (openssl_random_pseudo_bytes())
   https://github.com/paragonie/random_compat/issues/96

symfony/symfony (v2.7.23)
-------------------------

 * CVE-2017-16653: CVE-2017-16653: CSRF protection does not use different tokens for HTTP and HTTPS
   https://symfony.com/cve-2017-16653

 * CVE-2017-16654: CVE-2017-16654: Intl bundle readers breaking out of paths
   https://symfony.com/cve-2017-16654

 * CVE-2018-11386: CVE-2018-11386: Denial of service when using PDOSessionHandler
   https://symfony.com/cve-2018-11386

 * CVE-2017-16790: CVE-2017-16790: Ensure that submitted data are uploaded files
   https://symfony.com/cve-2017-16790

 * CVE-2018-11406: CVE-2018-11406: CSRF Token Fixation
   https://symfony.com/cve-2018-11406

 * CVE-2017-16652: CVE-2017-16652: Open redirect vulnerability on security handlers
   https://symfony.com/cve-2017-16652

 * CVE-2018-11385: CVE-2018-11385: Session Fixation Issue for Guard Authentication
   https://symfony.com/cve-2018-11385

willdurand/js-translation-bundle (1.2.2)
----------------------------------------

 * Fixed potential path traversal attack and remote code injection
   https://github.com/willdurand/BazingaJsTranslationBundle/releases/tag/v2.1.1
```

Now I've already tried to fix some of the vulnerabilities locally by "faking" versions in my local `composer.json`. 5 of the 8 dependencies have patch releases available:

```
"doctrine/annotations": "1.2.7 as 1.2.6"
"doctrine/common": "2.4.3 as 2.4.2"
"doctrine/orm": "2.4.8 as 2.4.7"
"dompdf/dompdf": "0.6.2 as 0.6.1"
"symfony/symfony": "2.7.48 as 2.7.23"
```

This however is no success, because now the console starts throwing errors:

```

  [Symfony\Component\Debug\Exception\ContextErrorException]
  Warning: Invalid argument supplied for foreach()


Exception trace:
 () at /var/www/vendor/akeneo/pim-community-dev/src/Pim/Bundle/ReferenceDataBundle/DependencyInjection/Compiler/RegisterConfigurationsPass.php:27
 Symfony\Component\Debug\ErrorHandler->handleError() at /var/www/vendor/akeneo/pim-community-dev/src/Pim/Bundle/ReferenceDataBundle/DependencyInjection/Compiler/RegisterConfigurationsPass.php:27
 Pim\Bundle\ReferenceDataBundle\DependencyInjection\Compiler\RegisterConfigurationsPass->process() at /var/www/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/Compiler.php:102
 Symfony\Component\DependencyInjection\Compiler\Compiler->compile() at /var/www/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:568
 Symfony\Component\DependencyInjection\ContainerBuilder->compile() at /var/www/app/bootstrap.php.cache:2673
 Symfony\Component\HttpKernel\Kernel->initializeContainer() at /var/www/app/bootstrap.php.cache:2451
 Symfony\Component\HttpKernel\Kernel->boot() at /var/www/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:62
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /var/www/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:114
 Symfony\Component\Console\Application->run() at /var/www/app/console:29
```

So it seems that some of the patch releases have BC breaking changes (I suspect `symfony/symfony`, since that jumps 24 patches at once).

So to sum it all up:

Since 1.7 still is under LTS, is Akeneo going to pick up these security vulnerabilities? I've read the reaction in the 1.5 post that it's very time consuming, but we're all left now with a PIM that has 8 dependencies with security issues. Perhaps you could fix them one by one and release 8 patch releases over time (1.7.22 -> 1.7.30)?

**Comment by jmleroux** [comment]
Hello @kanduvisla ,

thanks for raising this issue. I will forward it to our support team and we will handle it asap.

:+1:

**Comment by LaureBro** [comment]
Hello @kanduvisla, 

This issue has been fixed in the 1.7.26 version. If you update your PIM on this version, please remove the `APY\JsFormValidationBundle\APYJsFormValidationBundle` and `Pim\Bundle\JsFormValidationBundle\PimJsFormValidationBundle` bundles from your AppKernel.
Also you need to remove the routes from the `APYJsFormValidationBundle` bundle (this information will be added in our changelog shortly).

Note that an additional fix will be provided on 1.7.27 for Reference Data - the 1.7.27 is being currently released. 

Kind regards, 
Laure

**Comment by Doodoune** [comment]
Hello!

I close this issue resolve, feel free to reopen it if you still have this issue.

Regards,
Elodie
